app/models/validators/beyond_osra_establishment_validator.rb

```diff
@@ -0,0 +1,13 @@
+class BeyondOsraEstablishmentValidator < ActiveModel::EachValidator
+  OSRA_ESTABLISHMENT_DATE = Date.new(2012,04,01)
+  include Helpers
+
+  def validate_each(record, attribute, value)
+    if !valid_date? value
+      record.errors[attribute] << ('is not a valid date')
```

Comment:
Maybe we should make separate validators for `valid_date` too? Either way, still have to check here `if !valid_date? value` first. (I decided not to)